components/link_checker/src/lib.rs

```diff
@@ -46,39 +46,56 @@ impl LinkResult {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct LinkChecker {
+    /// Skip anchor checking for these URL prefixes
+    pub skip_anchor_prefixes: Vec<String>,
+}
+
+impl Default for LinkChecker {
+    fn default() -> LinkChecker {
+        LinkChecker { skip_anchor_prefixes: Vec::new() }
+    }
+}
+
 lazy_static! {
     // Keep history of link checks so a rebuild doesn't have to check again
     static ref LINKS: Arc<RwLock<HashMap<String, LinkResult>>> = Arc::new(RwLock::new(HashMap::new()));
 }
 
-pub fn check_url(url: &str) -> LinkResult {
```

Comment:
this would mean that this stays a function that is `pub fn check_url(url: &str, config: &LinkChecker) -> LinkResult`. Adding new options to the LinkChecker makes it available automatically

Reply:
So create link_checker should depend on crate config?